### PR TITLE
mcpe 1.0 chunkskylight

### PR DESCRIPTION
**Pull request by robske110**
This re-adds chunk skylight population support into mcpe-1.0

### CODE REVIEW

**Comment by robske110** [comment]
It would be nice if multiple people would test this. I am not sure if this works.

**Comment by dktapps** [comment]
This is mostly working fine now, apart from some lighting issues under trees that I can't make sense of.

File: src/pocketmine/level/format/generic/GenericChunk.php [hunk 1]
```diff
@@ -283,14 +284,14 @@ public function populateSkyLight(){
 				}
 
 				for(; $y > 0; --$y){
-					if(Block::$solid[$this->getBlockId($x, $y, $z)]){
+					if($this->getBlockId($x, $y, $z) !== Block::AIR){
```
Comment (jasonwynn10) [inline src/pocketmine/level/format/generic/GenericChunk.php]: what about leaves? they are transparent, so light should pass through them

Comment (robske110) [inline src/pocketmine/level/format/generic/GenericChunk.php]: No, they diffuse skylight. This was and will be a very basic population.

**Pull request merged**